### ISSUE TITLE
False negative in clang-tidy check readability-convert-member-functions-to-static with nested classes

### ISSUE THREAD

**Issue by njames93** [llvm/llvm-project#56765]
The check gets confused when a method has a nested class.

```c++
struct Foo {
  // No warning emitted.
  void canBeStatic() {
    struct Nested {
      int Z;
      int getZ() { return Z; }
    };
  }
};
```
https://godbolt.org/z/vW77fseY6
